apps/next/src/memlab/lib/baseline.ts

```diff
@@ -0,0 +1,316 @@
+    currentRetainedSize: result.totalRetainedSize,
+    baselineRetainedSize: baselineEntry.totalRetainedSize,
+    retainedSizeDelta,
+    isRegression,
+    isImprovement,
+  };
+}
+
+/**
+ * Format baseline comparison for console output
+ */
+export function formatBaselineComparison(
+  comparison: BaselineComparison,
+): string {
+  const lines: string[] = [];
+
+  const leakIcon =
+    comparison.leakCountDelta > 0
+      ? '📈'
+      : comparison.leakCountDelta < 0
+        ? '📉'
+        : '➡️';
+  const sizeIcon =
+    comparison.retainedSizeDelta > 0
+      ? '📈'
+      : comparison.retainedSizeDelta < 0
+        ? '📉'
+        : '➡️';
+
+  lines.push(`\n📊 Baseline Comparison for ${comparison.component}:`);
+  lines.push(
+    `   ${leakIcon} Leaks: ${comparison.currentLeakCount} (baseline: ${comparison.baselineLeakCount}, delta: ${comparison.leakCountDelta >= 0 ? '+' : ''}${comparison.leakCountDelta})`,
+  );
+  lines.push(
+    `   ${sizeIcon} Retained: ${formatBytes(comparison.currentRetainedSize)} (baseline: ${formatBytes(comparison.baselineRetainedSize)}, delta: ${comparison.retainedSizeDelta >= 0 ? '+' : ''}${formatBytes(Math.abs(comparison.retainedSizeDelta))})`,
+  );
+
+  if (comparison.isRegression) {
+    lines.push('   ⚠️ REGRESSION DETECTED');
+  } else if (comparison.isImprovement) {
+    lines.push('   ✅ IMPROVEMENT DETECTED');
+  } else {
+    lines.push('   ➡️ No significant change');
+  }
+
+  return lines.join('\n');
+}
+
+/**
+ * Generate baseline report for multiple components
+ */
+export function generateBaselineReport(
+  comparisons: BaselineComparison[],
+): string {
+  const lines: string[] = [];
+
+  lines.push(`\n${'='.repeat(60)}`);
+  lines.push('📊 BASELINE COMPARISON REPORT');
+  lines.push('='.repeat(60));
+
+  const regressions = comparisons.filter((c) => c.isRegression);
+  const improvements = comparisons.filter((c) => c.isImprovement);
+  const unchanged = comparisons.filter(
+    (c) => !(c.isRegression || c.isImprovement),
+  );
+
+  if (regressions.length > 0) {
+    lines.push(`\n⚠️ Regressions: ${regressions.length}`);
+    for (const r of regressions) {
+      lines.push(
+        `   - ${r.component}: +${r.leakCountDelta} leaks, +${formatBytes(Math.abs(r.retainedSizeDelta))}`,
+      );
+    }
+  }
+
+  if (improvements.length > 0) {
+    lines.push(`\n✅ Improvements: ${improvements.length}`);
+    for (const i of improvements) {
+      lines.push(
+        `   - ${i.component}: ${i.leakCountDelta} leaks, ${formatBytes(Math.abs(i.retainedSizeDelta))}`,
+      );
+    }
+  }
+
+  if (unchanged.length > 0) {
+    lines.push(`\n➡️ Unchanged: ${unchanged.length}`);
+    for (const u of unchanged) {
+      lines.push(`   - ${u.component}`);
+    }
+  }
+
+  lines.push(`\n${'='.repeat(60)}`);
+
+  return lines.join('\n');
+}
+
+/**
+ * Get current git commit hash
+ */
+export function getGitCommitHash(): string | undefined {
+  try {
+    const { execSync } = require('node:child_process');
```

Comment:
checking my understanding: why is require needed here. Could this get converted to a regular import?

```suggestion
// at the top of the file: 
import { execSync } from 'node:child_process';
```